sklearn/ensemble/tests/test_gradient_boosting_loss_functions.py

```diff
@@ -34,23 +34,28 @@ def test_binomial_deviance():
     assert_almost_equal(bd(np.array([1.0, 0.0, 0.0]),
                            np.array([100.0, -100.0, -100.0])), 0)
 
-    # check if same results as alternative definition of deviance (from ESLII)
-    def alt_dev(y, pred):
-        return np.mean(np.logaddexp(0.0, -2.0 * (2.0 * y - 1) * pred))
+    # check if same results as alternative definition of deviance, from ESLII
+    # Eq. (10.18): -loglike = log(1 + exp(-2*z*f))
+    # Note:
+    # - We use y = {0, 1}, ESL (10.18) uses z in {-1, 1}, hence y=2*y-1
+    # - ESL 2*f = pred_raw, hence the factor 2 of ESL disappears.
+    # - Deviance = -2*loglike + .., hence a factor of 2 in front.
```

Comment:
thanks for the comments. I was a bit confused by ESL because they say

"or equivalently the deviance is -loglike = ... (10.18)" but it seems like it's missing a *2 factor in the book

Reply:
*de rien*. First, I was confused my self and thus wrote an a4 page of log loss formulae:nerd_face: